src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc

```diff
@@ -204,7 +204,14 @@ void FileWatcherCertificateProvider::ForceUpdate() {
        pem_key_cert_pairs_ != *pem_key_cert_pairs);
   if (identity_cert_changed) {
     if (pem_key_cert_pairs.has_value()) {
-      pem_key_cert_pairs_ = std::move(*pem_key_cert_pairs);
+      absl::StatusOr<bool> matched_or = PrivateKeyAndCertificateMatch(*pem_key_cert_pairs);
+      if (matched_or.ok() && *matched_or){
+        pem_key_cert_pairs_ = std::move(*pem_key_cert_pairs);
+      }else{
+        gpr_log(GPR_ERROR,
+              "PrivateKeyAndCertificateMatch failed, skip the update. Error message: %d",
```

Comment:
Why are we using `%d` here? Shouldn't it be `%s`? Is this line of code actually executed by your test?

Reply:
Yeah, that code is reached when the new credentials pair are a mismatch or there is some other error with PrivateKeyAndCertificateMatch. The error message will be (null) if its just a mismatch

Reply:
That was a mistake, I'll fix it